src/ck-ldb/tree_level.ci

```diff
@@ -0,0 +1,35 @@
+module tree_level {
+
+  message LBStatsMsg_1 {
+    int pe_ids[];
+    float bgloads[];
+    float speeds[];
+    unsigned int obj_start[];
+    float oloads[];
+    unsigned int order[];
+  };
+
+  message LLBMigrateMsg {
+    int num_incoming[];
+    int obj_start[];
+    int to_pes[];
+  };
+
+  message SubtreeLoadMsg {
+  };
+
+  message SubtreeMigrateDecisionMsg {
+    // FIXME? I don't think this layout is ideal, but I don't know if ci files
```

Comment:
We should support arrays of custom structs in messages, particularly if those types are declared as PUPbytes

Reply:
By that I mean I think we already do support it

Reply:
Thanks, I will have to revisit that. But I think we are going to split this into multiple PRs so you should remember to move this comment to the TreeLB PR.